src/pydvl/utils/caching/memcached.py

```diff
@@ -0,0 +1,205 @@
+
+    PYMEMCACHE_INSTALLED = True
+except ImportError:
+    PYMEMCACHE_INSTALLED = False
+
+from .base import CacheBackend
+
+__all__ = ["MemcachedClientConfig", "MemcachedCacheBackend"]
+
+PICKLE_VERSION = 5  # python >= 3.8
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class MemcachedClientConfig:
+    """Configuration of the memcached client.
+
+    Args:
+        server: A tuple of (IP|domain name, port).
+        connect_timeout: How many seconds to wait before raising
+            `ConnectionRefusedError` on failure to connect.
+        timeout: Duration in seconds to wait for send or recv calls
+            on the socket connected to memcached.
+        no_delay: If True, set the `TCP_NODELAY` flag, which may help
+            with performance in some cases.
+        serde: Serializer / Deserializer ("serde"). The default `PickleSerde`
+            should work in most cases. See [pymemcache.client.base.Client][]
+            for details.
+    """
+
+    server: Tuple[str, int] = ("localhost", 11211)
+    connect_timeout: float = 1.0
+    timeout: float = 1.0
+    no_delay: bool = True
+    serde: PickleSerde = PickleSerde(pickle_version=PICKLE_VERSION)
+
+
+class MemcachedCacheBackend(CacheBackend):
+    """Memcached cache backend for the distributed caching of functions.
+
+    Implements the CacheBackend interface for a memcached based cache.
+    This allows sharing evaluations across processes and nodes in a cluster.
+    You can run memcached as a service, locally or remotely,
+    see [Setting up the cache](#setting-up-the-cache)
+
+    Args:
+        config: Memcached client configuration.
+
+    Attributes:
+        config: Memcached client configuration.
+        client: Memcached client instance.
+
+    ??? Examples
+        ``` pycon
+        >>> from pydvl.utils.caching.memcached import MemcachedCacheBackend
+        >>> cache_backend = MemcachedCacheBackend()
+        >>> cache_backend.clear()
+        >>> value = 42
+        >>> cache_backend.set("key", value)
+        >>> cache_backend.get("key")
+        42
+        ```
+
+        ``` pycon
+        >>> from pydvl.utils.caching.memcached import MemcachedCacheBackend
+        >>> cache_backend = MemcachedCacheBackend()
+        >>> cache_backend.clear()
+        >>> value = 42
+        >>> def foo(x: int):
+        ...     return x + 1
+        ...
+        >>> wrapped_foo = cache_backend.wrap(foo)
+        >>> wrapped_foo(value)
+        43
+        >>> wrapped_foo.stats.misses
+        1
+        >>> wrapped_foo.stats.hits
+        0
+        >>> wrapped_foo(value)
+        43
+        >>> wrapped_foo.stats.misses
+        1
+        >>> wrapped_foo.stats.hits
+        1
+        ```
+    """
+
+    def __init__(self, config: MemcachedClientConfig = MemcachedClientConfig()) -> None:
+        """Initialize memcached cache backend.
+
+        Args:
+            config: Memcached client configuration.
+        """
+        if not PYMEMCACHE_INSTALLED:
```

Comment:
I know it's trivial, but maybe we should use pytest-mock to test this? (The pattern probably repeats elsewhere)
```python

from pydvl.utils.caching.memcached import MemcachedBackend

def test_without_pymemcache(mocker):
    mocker.patch.dict('sys.modules', {'pymemcache': None})
    with pytest.raises(ModuleNotFoundError):
        backend = MemcachedBackend()

```